components/authn-service/tokens/mock/mock.go

```diff
@@ -127,6 +127,22 @@ func (m *mock) DeleteToken(ctx context.Context, id string) error {
 	return nil
 }
 
+func (m *mock) PurgeProject(ctx context.Context, projectID string) error {
+	for _, tok := range m.tokens {
+		for i, v := range tok.Projects {
+			if v == projectID {
+				tok.Projects = append(tok.Projects[:i], tok.Projects[i+1:]...)
+				break
+			}
+		}
+		_, err := m.UpdateToken(ctx, tok.ID, tok.Description, tok.Active, tok.Projects)
```

Comment:
Don't need to call `UpdateToken` if the project was never encountered in the above loop... oh, but we're in a mock file, so it's fine.